### ISSUE TITLE
Quick way to switch x axis between time and distance

### ISSUE THREAD

**Issue by tobiaslj** [Runalyze/Runalyze#1698]
Add an icon to easily switch the x-axis on plots between time and distance. The preference should be stored together with the activity and user preference should only be used as default for new activities.

**Comment by mipapo** [comment]
Saving user preferences will be handled in #1610
@laufhannes I think we won't add this functionality until d3js graphs?

**Comment by laufhannes** [comment]
#1610 is about general graphs - not about the activity view.

I don't see why to save preferences for single activities. There are thousands of activities per user, most of them never viewed again, and 99 % of them would have same preferences anyway.

A little icon/link to switch between time and distance as x-axis is a good idea. But @mipapo is right: This is better done after refactoring the graphs, i.e. switching from flot to d3js.

**Comment by tobiaslj** [comment]
@laufhannes I agree that many activities will never be viewed again but if you do serious training there will always be activities that are more important than other. Activities that in one way or another will give you information about your current status and comparing them will help you analyse your performance.
In most of the cases plots over distance is what you want, but sometimes time is important. Mostly related to HR. It can't be that difficult to have some options stored per exercise in the database. If nothing is set the user preference is used.

**Comment by mipapo** [comment]
Well it's not difficult to add this functionality, but it makes no sense to add this. 
You set once your standard x axis to distance or time. If you're viewing an activity we will may add in the future the functionality to easily switch the x-axis with one click. 
If you look again and again at one special activity you'll have to click on the button so see the distance or time. 
We will keep this in mind for the refactoring of the graphs, but for now we won't change anything. 

(One thing that crosses my mind is that we can add a third option in the user settings, which combines time and distance on the x axis for the graphs)
